### PR TITLE
Allow extra include/exclude filters in project searches

### LINKED ISSUE #795: Include/Exclude field for in-project find dialog
Currently we have only include field. It would be nice to have exclude field also

Dev edit:

Added want for "Include" support as well. This doesn't work with Projects.

### PR DESCRIPTION
**Pull request by ssigwart**
- I think this closes #795

image

### CODE REVIEW

File: src/projects/koProject.p.py [hunk 1]
```diff
@@ -1020,14 +1020,22 @@ def _getImportConfig(self, recursive=0, type="makeFlat"):
...
         """Generate all contained local paths."""
         from os.path import join
         
         prefset = self.get_prefset()
         base_dir = self.get_liveDirectory()
         excludes = [p for p in prefset.getStringPref("import_exclude_matches").split(';') if p]
+        if extraExcludes != None:
+            for extraPattern in extraExcludes.split(';'):
+                if extraPattern:
+                    excludes.append(extraPattern);
```
Comment (cgchoffman) [inline src/projects/koProject.p.py]: Just get you to remove this

File: src/projects/koProject.p.py [hunk 2]
```diff
@@ -1020,14 +1020,22 @@ def _getImportConfig(self, recursive=0, type="makeFlat"):
...
         includes = [p for p in prefset.getStringPref("import_include_matches").split(';') if p]
+        if extraIncludes != None:
+            for extraPattern in extraIncludes.split(';'):
+                if extraPattern:
+                    includes.append(extraPattern);
```
Comment (cgchoffman) [inline src/projects/koProject.p.py]: And this

**Comment by ssigwart** [comment]
I'm never going to get used to that about Python.  Fixed.

**Comment by cgchoffman** [comment]
I believe in you @ssigwart :|

**Pull request merged**